pkg/plugins/mco/mco_plugin.go

```diff
@@ -0,0 +1,200 @@
+	glog.Info("mco-plugin OnNodeStateChange()")
+	switchdevConfigured = false
+	for _, iface := range new.Spec.Interfaces {
+		if iface.EswitchMode == sriovnetworkv1.ESWITCHMODE_SWITCHDEV {
+			switchdevConfigured = true
+		}
+	}
+
+	var update, remove bool
+	if update, remove, err = writeSwitchdevConfFile(new); err != nil {
+		glog.Errorf("mco-plugin OnNodeStateChange():fail to update switchdev.conf file: %v", err)
+		return
+	}
+	if remove {
+		needDrain = true
+		return
+	}
+	if update {
+		if _, err = os.Stat(switchdevUnitPath); err != nil {
+			if os.IsNotExist(err) {
+				glog.Info("mco-plugin OnNodeStateChange(): the latest MachineConfig has not been applied")
+				needDrain = true
+				err = nil
+				return
+			}
+			return
+		}
+		// node is already in the offload MCP
+		glog.Info("mco-plugin OnNodeStateChange(): need reboot node to use the up-to-date switchdev.conf")
+		needDrain = true
+		needReboot = true
+		return
+	}
+	return
+}
+
+// Apply config change
+func (p *McoPlugin) Apply() error {
+	glog.Info("mco-plugin Apply()")
+	node, err := kubeclient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	label := nodeLabelPrefix + controllers.HwOffloadNodeLabel
+	if switchdevConfigured {
+		if _, ok := node.Labels[label]; !ok {
+			glog.Infof("Move node %s into HW offload MachineConfigPool", node.Name)
+			mergePatch, _ := json.Marshal(map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"labels": map[string]interface{}{
+						label: "",
+					},
+				},
+			})
+			kubeclient.CoreV1().Nodes().Patch(context.Background(), nodeName, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+			return nil
+		}
+		glog.Infof("Node %s is already in HW offload MachineConfigPool", node.Name)
+		return nil
+	}
+
+	if _, ok := node.Labels[label]; ok {
+		glog.Infof("Remove node %s from HW offload MachineConfigPool", node.Name)
+		mergePatch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]interface{}{
+					label: nil,
+				},
+			},
+		})
+		kubeclient.CoreV1().Nodes().Patch(context.Background(), nodeName, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+		return nil
+	}
+	glog.Infof("Node %s is not in HW offload MachineConfigPool", node.Name)
+	return nil
+}
+
+func writeSwitchdevConfFile(newState *sriovnetworkv1.SriovNetworkNodeState) (update, remove bool, err error) {
+	_, err = os.Stat(switchDevConfPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			glog.V(2).Infof("writeSwitchdevConfFile(): file not existed, create it")
+			_, err = os.Create(switchDevConfPath)
+			if err != nil {
+				glog.Errorf("writeSwitchdevConfFile(): fail to create file: %v", err)
+				return
+			}
+		} else {
+			return
+		}
+	}
+	newContent := ""
+	for _, iface := range newState.Spec.Interfaces {
+		if iface.EswitchMode == sriovnetworkv1.ESWITCHMODE_SWITCHDEV {
```

Comment:
Add a condition that the interface support changing eswitchMode, the state interface has eswitchMode !=""

Reply:
Is there a way to check whether an interface support changing eswitchMode or not? EswitchMode in the NodeState can only be either 'switchdev' or ''.